spec/lib/msf/ui/console/command_dispatcher/auxiliary_spec.rb

```diff
@@ -1,23 +1,734 @@
+          'Checking for target 192.0.2.1:3000 with normalized datastore value 3.5',
+          'Cleanup for target 192.0.2.1:3000',
+          '192.0.2.1:3000 - Check failed: The state could not be determined.',
+          'Checking for target 192.0.2.2:3000 with normalized datastore value 3.5',
+          'Cleanup for target 192.0.2.2:3000',
+          '192.0.2.2:3000 - Check failed: The state could not be determined.',
+          'Cleanup for target 192.0.2.1 192.0.2.2:3000'
+        ]
+
+        expect(@combined_output).to match_array(expected_output)
+      end
+
+      it 'normalizes the datastore before running' do
+        current_mod.datastore['RHOSTS'] = '192.0.2.1 192.0.2.2'
+        current_mod.datastore.store('FloatValue', '5.0')
+        subject.cmd_check
+        expected_output = [
+          'Checking for target 192.0.2.1:3000 with normalized datastore value 5.0',
+          'Cleanup for target 192.0.2.1:3000',
+          '192.0.2.1:3000 - Check failed: The state could not be determined.',
+          'Checking for target 192.0.2.2:3000 with normalized datastore value 5.0',
+          'Cleanup for target 192.0.2.2:3000',
+          '192.0.2.2:3000 - Check failed: The state could not be determined.',
+          'Cleanup for target 192.0.2.1 192.0.2.2:3000'
+        ]
+
+        expect(@combined_output).to match_array(expected_output)
+      end
+
+      it 'supports inline options' do
+        pending('cmd_check does not support inline values yet')
+        current_mod.datastore.store('FloatValue', '5.0')
+        subject.cmd_check('RHOSTS=192.0.2.5', 'FloatValue=10.0')
+        expected_output = [
+          'Checking for target 192.0.2.5:3000 with normalized datastore value 10.0',
+          'Cleanup for target 192.0.2.5:3000',
+          '192.0.2.5:3000 - Check failed: The state could not be determined.',
+          'Cleanup for target 192.0.2.5:3000'
+        ]
+
+        expect(@combined_output).to match_array(expected_output)
+      end
+
+      it 'supports multiple inlined RHOST values' do
+        pending('pending as inline module options are evaluated too late, and the module is therefore treated as a scanner')
+        current_mod.datastore.store('FloatValue', '5.0')
+        subject.cmd_check('RHOSTS=192.0.2.5 192.0.2.6', 'FloatValue=10.0')
+        expected_output = [
+          'Checking for target 192.0.2.5:3000 with normalized datastore value 10.0',
+          'Cleanup for target 192.0.2.5:3000',
+          '192.0.2.5:3000 - Check failed: The state could not be determined.',
+          'Checking for target 192.0.2.6:3000 with normalized datastore value 5.0',
+          'Cleanup for target 192.0.2.6:3000',
+          '192.0.2.6:3000 - Check failed: The state could not be determined.',
+          'Cleanup for target 192.0.2.5 192.0.2.6:3000'
+        ]
+
+        expect(@combined_output).to match_array(expected_output)
+      end
+
+      it 'ignores the -j flag, and the module is not run as a job' do
```

Comment:
is the `-j` flag being ignored a bug or feature?

Reply:
I think it's maybe missing functionality, you can trace the usage of `jobify` through the `cmd_run` and see that it only honors the `-j` flag with scanners and modules and when rhosts is missing:
https://github.com/rapid7/metasploit-framework/blob/6dd298ebb76a1617e24da5e4c73e43a46b226a23/lib/msf/ui/console/command_dispatcher/auxiliary.rb#L46

The flags are entirely ignored/not parsed in the check method:
https://github.com/rapid7/metasploit-framework/blob/6dd298ebb76a1617e24da5e4c73e43a46b226a23/lib/msf/ui/console/module_command_dispatcher.rb#L123